python/cudf/cudf/core/column/datetime.py

```diff
@@ -433,14 +433,11 @@ def can_cast_safely(self, to_dtype: Dtype) -> bool:
     def _make_copy_with_na_as_null(self):
         """Return a copy with NaN values replaced with nulls."""
         null = column_empty_like(self, masked=True, newsize=1)
+        na_value = np.datetime64("nat", self.time_unit)
```

Comment:
I probably should have explained this more in the PR description. I made `default_na_value` an internal method (prefixed with an underscore) in this PR and plan to remove it entirely when we remove the `as_matrix` and `as_gpu_matrix` methods for good. The reason is that not only do we not use this variable consistently in places where methods should have a default na value, there are also many methods that need different defaults (or in some cases shouldn't use a default at all). The variable is more confusing than helpful, and over time many of the relevant APIs will be phased out anyway because it turns out that more often than not we don't want a default.

Reply:
Thanks for the explanation - with that context, the explicit assignment of `na_value` makes sense, and as you said we can open a PR later on to phase out the use of `_default_na_value`.